### ISSUE TITLE
Group membership fails to apply when using keep_bots attribute

### ISSUE THREAD

**Issue by SachinKSingh28** [gitlabform/gitlabform#553]
## Description

Following fails to apply group membership - 

```yaml
config_version: 3
gitlab:
  ssl_verify: true
  timeout: 2
projects_and_groups:
  "GroupA/GroupB/*":
    group_members:
      keep_bots: true
      enforce: true
      groups:
        xx/users-reporter:
          group_access: reporter
        yy/users-guest:
          group_access: guest
```

Use following command to replicate this issue - 

```sh
gitlabform -c above-config.yaml GroupA/GroupB/GroupC -os group_settings,group_members
```

I'm getting below error message - 

```
🏗 GitLabForm version: 3.6.0 = the latest stable 😊
Warning: Connected as non-admin. You may encounter permission issues.
>>> Getting requested groups or projects...
:: # of groups to process: 1
:: # of projects to process: 0
@ (1/1) Processing group: GroupA/GroupB/GroupC 
Warning: Error occurred while processing group GroupA/GroupB/GroupC, exception:

'bool' object is not subscriptable
:: # of groups processed successfully: 0
:: # of projects processed successfully: 0
:: # of groups failed: 1 
:: Failed group 1: GroupA/GroupB/GroupC
```

On debugging it seems the user list for group membership had following item - 

```
keep_bots: true
```

### Gitlabform version 

```sh
% gitlabform --version                                                                                  
🏗 GitLabForm version: 3.6.0 = the latest stable 😊
```

### Gitlab version

GitLab Enterprise Edition 16.1.0-pre [c5ac07d12dc](https://gitlab.com/gitlab-org/gitlab/-/commits/c5ac07d12dc)

**Comment by SachinKSingh28** [comment]
> > Use following command to replicate this issue -
> > ```
> > gitlabform -c above-config.yaml GroupA/GroupB/GroupC -os group_settings,group_members
> > ```
> 
> What does the `-os` parameter do? I'm not familiar with it. Not sure if I'm forgetting something or missed a new feature release recently.

@amimas -os is only-sections allowing one to just apply specific configuration blocks. We use this to fill the gap in recursive group settings apply. Ref - https://github.com/gitlabform/gitlabform/issues/509#issuecomment-1466585755

**Comment by amimas** [comment]
Thanks for clarifying. Not sure if I missed it or not, but couldn't find that option in the docs. But, I found the option in the code:

https://github.com/gitlabform/gitlabform/blob/24bdac6d8fdb0dbde74e82c3c425f805339741e6/gitlabform/__init__.py#L253-L260

Can you try it without that argument? Use `ALL_DEFINED` instead:

```
gitlabform -c above-config.yaml ALL_DEFINED
```

I know there were issues reported previously where things are processed properly with the above command but doesn't work if ran it differently. Not sure if that's been fixed. Curious if it's a similar bug like that.

**Comment by long-wan-ep** [comment]
We are also seeing this error after upgrading to 3.6.0 and adding `keep_bots: true` to our `group_members/members` blocks. It only seems to happen to nested groups(ie. `GroupA/GroupB/*`) and works fine for top level groups(ie. `GroupA/*`).

Also, when we ran gitlabform with `--noop`, everything was working and the generated yaml output looked correct. When we ran without `--noop` we got the error and the `group_member` blocks were missing in the generated output.

**Comment by long-wan-ep** [comment]
Running with the `--debug` flag, this is the stack trace I get:
```
'bool' object is not subscriptable
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/gitlabform/__init__.py", line 398, in run
    self.group_processors.process_entity(
  File "/opt/homebrew/lib/python3.11/site-packages/gitlabform/processors/__init__.py", line 31, in process_entity
    processor.process(
  File "/opt/homebrew/lib/python3.11/site-packages/gitlabform/processors/util/decorators.py", line 42, in method_wrapper
    return method(self, project_and_group, SafeDict(configuration), *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/gitlabform/processors/abstract_processor.py", line 58, in process
    self._process_configuration_with_retries(
  File "/opt/homebrew/lib/python3.11/site-packages/gitlabform/processors/abstract_processor.py", line 100, in _process_configuration_with_retries
    raise e
  File "/opt/homebrew/lib/python3.11/site-packages/gitlabform/processors/abstract_processor.py", line 86, in _process_configuration_with_retries
    self._process_configuration(
  File "/opt/homebrew/lib/python3.11/site-packages/gitlabform/processors/group/group_members_processor.py", line 39, in _process_configuration
    self._process_users(
  File "/opt/homebrew/lib/python3.11/site-packages/gitlabform/processors/group/group_members_processor.py", line 162, in _process_users
    access_level = users_to_set_by_username[user]["access_level"]
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
```

I'm not familiar with the code, but I'm thinking this [block](https://github.com/gitlabform/gitlabform/blob/main/gitlabform/processors/group/group_members_processor.py#L54C18-L57) may need a pop for `keep_bots`?

**Comment by gdubicki** [comment]
Released in [v3.7.0](https://github.com/gitlabform/gitlabform/releases/tag/v3.7.0) a moment ago. Thank you again for your contribution, @SachinKSingh28! 🤝
